### PR TITLE
77486 - stub creation for aio for base cd

### PR DESCRIPTION
**Pull request by sumod13943**
Following PBI's are covered as part of this PR

1) 77486 Enabler - POS : Stub creation for AIO for base CD
2) 77487 Enabler - POS : Stub creation for AIO Update

It includes modification to the existing mock to handle the scenarios of AIO and an endpoint for product version is added to the controller for AIO update.

### CODE REVIEW

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs [hunk 1]
```diff
@@ -90,10 +141,11 @@ private ExchangeSetServiceResponse GetEssResponse(string responseId)
         private CreateBatchRequest CreateBatchRequestModel(bool isPostProductIdentifiersRequest)
         {
             PosTestCase currentTestCase = _mockService.GetCurrentPOSTestCase(_homeDirectoryPath);
+            string batchType;
 
-            string batchType = currentTestCase != PosTestCase.ValidProductIdentifiers
-                ? currentTestCase.ToString()
-                : isPostProductIdentifiersRequest ? Batch.EssFullAvcsZipBatch.ToString() : Batch.EssUpdateZipBatch.ToString();
+            batchType = currentTestCase != PosTestCase.ValidProductIdentifiers
```
Comment (DanOwenUKHO) [inline UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs:146]: Remove line 144 and declare directly here

Split out conditional expression for if/else statements to improve readability

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs [hunk 2]
```diff
@@ -78,6 +91,44 @@ public ExchangeSetServiceResponse CreateExchangeSetForPostProductIdentifier(stri
...
+
+                BatchResponse createBatchResponse = _fssService.CreateBatch(batchRequest.Attributes, _homeDirectoryPath);
+                string productVersion = $"productVersion-{item.ProductName}-{item.EditionNumber}-{item.UpdateNumber}";
+
+                if (!string.IsNullOrEmpty(createBatchResponse.BatchId.ToString()))
+                {
+                    string path = Path.Combine(Environment.CurrentDirectory, @"Data", createBatchResponse.BatchId.ToString());
+                    foreach (string fileName in Directory.GetFiles(path))
+                    {
+                        FileInfo file = new(fileName);
```
Comment (DanOwenUKHO) [inline UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs:119]: Generate a FileInfo object from the fileName, but then only use it to access the fileName (line 120). Perhaps easier to just use the fileName directy

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs [hunk 3]
```diff
@@ -78,6 +91,44 @@ public ExchangeSetServiceResponse CreateExchangeSetForPostProductIdentifier(stri
             return null;
         }
 
+        public ExchangeSetServiceResponse CreateExchangeSetForPostProductVersion(List<ProductVersionRequest> productVersionsRequest)
+        {
+            CreateBatchRequest batchRequest;
+
+            foreach (ProductVersionRequest? item in productVersionsRequest)
```
Comment (DanOwenUKHO) [inline UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs:98]: Why nullable

**Review by DanOwenUKHO — CHANGES_REQUESTED** [review]
A few minor comments.

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/MockService.cs [hunk 4]
```diff
@@ -28,6 +29,33 @@ public PosTestCase GetCurrentPOSTestCase(string homeDirectoryPath)
...
+
+        public void UpdateAIOTestCase(AioTestCase aioTestCase, string homeDirectoryPath)
+        {
+            string destPath = Path.Combine(homeDirectoryPath, currentAioTestFileName);
+
+            if (!File.Exists(destPath))
+            {
+                File.Create(destPath).Close();
+            }
+            File.WriteAllText(destPath, "");
```
[the remark this answers is not in the document]
Comment (sumod13943) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/MockService.cs:55]: Removed this line as the "File.WriteAllText" will replace all the text with the one that is passed.

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs (hunk 1 above)
Comment (sumod13943) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs:146]: Resolved

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs (hunk 2 above)
Comment (sumod13943) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs]: The filename used in the Fileinfo object is in fact the file path. Therefore renamed the variable.

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs (hunk 3 above)
Comment (sumod13943) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs]: Removed nullable

**Review by shailen14599 — APPROVED** [review]
LGTM

**Review by DanOwenUKHO — APPROVED** [review]
LGTM

**Pull request merged**